pandas/tests/groupby/test_groupby.py

```diff
@@ -361,6 +363,119 @@ def f3(x):
         df2.groupby("a").apply(f3)
 
 
+def test_groupby_indices_error():
+    # GH 26860
+    # Test if DataFrame Groupby builds gb.indices
+    dt = pd.to_datetime(["2018-01-01", "2018-02-01", "2018-03-01"])
+    df = DataFrame(
+        {
+            "a": Series(list("abc")),
+            "b": Series(dt, dtype="category"),
+            "c": pd.Categorical.from_codes([-1, 0, 1], categories=[0, 1]),
+        }
+    )
+
+    df.groupby(["a", "b"]).indices
+
+
+@pytest.mark.parametrize(
+    "gb_cols",
+    [
+        "int_series",
+        "int_series_cat",
+        "float_series",
+        "float_series_cat",
+        "dt_series",
+        "dt_series_cat",
+        "period_series",
+        "period_series_cat",
+        [
+            "int_series",
+            "int_series_cat",
+            "float_series",
+            "float_series_cat",
+            "dt_series",
+            "dt_series_cat",
+            "period_series",
+            "period_series_cat",
+        ],
+    ],
+)
+def test_groupby_indices_output(gb_cols):
+    # GH 26860
+    # Test if DataFrame Groupby builds gb.indices correctly.
+    if isinstance(gb_cols, str):
+        gb_cols = [gb_cols]
+
+    cols = [
+        "int_series",
+        "int_series_cat",
+        "float_series",
+        "float_series_cat",
+        "dt_series",
+        "dt_series_cat",
+        "dttz_series",
+        "dttz_series_cat",
+        "period_series",
+        "period_series_cat",
+    ]
+
+    int_series = Series([1, 2, 3])
+    dt_series = pd.to_datetime(["2018Q1", "2018Q2", "2018Q3"])
+    dttz_series = dt_series.tz_localize("Europe/Berlin")
+    df = DataFrame(
+        data={
+            "int_series": int_series,
+            "int_series_cat": int_series.astype("category"),
+            "float_series": int_series.astype("float"),
+            "float_series_cat": int_series.astype("float").astype("category"),
+            "dt_series": dt_series,
+            "dt_series_cat": dt_series.astype("category"),
+            "dttz_series": dttz_series,
+            "dttz_series_cat": dttz_series.astype("category"),
+            "period_series": dt_series.to_period("Q"),
+            "period_series_cat": dt_series.to_period("Q").astype("category"),
+        },
+        columns=cols,
+    )
+
+    def dt_to_ts(elems):
```

Comment:
personal preference but I'm not sure defining these, while for sure it avoids code duplication, makes this easier to grok

Reply:
since you only use them twice

Reply:
Yup.  They are holdovers from the previous attempt at this test.  Not necessary now.